l10n_th_tin_service/models/res_partner.py

```diff
@@ -0,0 +1,196 @@
+# Copyright 2020 Poonlap V. <poonlap@tanabutr.co.th>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+import logging
+import pprint
+import re
+from collections import OrderedDict
+from logging import DEBUG, INFO
+from os import path
+from pathlib import Path
+
+import requests
+from requests import Session
+from zeep import Client, Transport, helpers
+
+from odoo import _, api, models
+
+_logger = logging.getLogger(__name__)
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+    tin_web_service_url = (
+        "https://rdws.rd.go.th/serviceRD3/checktinpinservice.asmx?wsdl"
+    )
+    vat_web_service_url = "https://rdws.rd.go.th/serviceRD3/vatserviceRD3.asmx?wsdl"
+
+    @staticmethod
+    def check_rd_tin_service(tin):
```

Comment:
This should be a private method.

```suggestion
    @staticmethod
    def _check_rd_tin_service(tin):
```